src/grz_pydantic_models/submission/metadata/v1.py

```diff
@@ -781,19 +781,24 @@ def validate_vcf_file_exists(self):
         return self
 
     @model_validator(mode="after")
-    def validate_fastq_file_exists(self):  # noqa: C901
+    def validate_sequencing_file_exists(self):  # noqa: C901
         """
-        Check if there is a FASTQ file
+        Check if there is a FASTQ or BAM file (depending on library type)
         """
         for lab_datum in self.lab_data:
             if not lab_datum.has_sequence_data():
                 # Skip if no sequence data is present
                 continue
             fastq_files = lab_datum.sequence_data.list_files(FileType.fastq)
+            bam_files = lab_datum.sequence_data.list_files(FileType.bam)
 
-            if len(fastq_files) == 0:
-                raise ValueError("No FASTQ file found!")
-            elif lab_datum.sequencing_layout == SequencingLayout.paired_end:
+            if not lab_datum.library_type.endswith("_lr") and len(bam_files) > 0:
+                raise ValueError("Only long-read lab data can have BAM files!")
+
+            if len(fastq_files) + len(bam_files) == 0:
+                raise ValueError("No FASTQ or BAM files found!")
```

Comment:
I'd be careful with the error message as well, so users don't think non _lr cases support BAM.